src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -713,9 +716,14 @@ private void AddEasyHandle(RequestCompletionSource state)
                         throw new HttpRequestException(SR.net_http_client_execution_error,
                             GetCurlException(result, true));
                     }
+                    state.SessionHandle = _multiHandle;
+                    if (null == _multiHandle.FdSet)
+                    {
+                        _multiHandle.FdSet = new HashSet<int>();
+                        PollFunction(_multiHandle);
```

Comment:
What happens when the last handle is removed from the FdSet? We can stop polling then, right?  At which point we'd need to then restart polling when FdSet count goes up from 0 to 1 again, right?